### DISCUSSION TITLE
Teradata Tools and Utilities hooks and operators

### DISCUSSION THREAD

**Discussion by flolas** [apache/airflow#18831] [Ideas]
**Description**
I'm working in operators and hooks for Teradata Database and I want to share with the Airflow Community. There are several tools(C tools) for loading/extract data in Teradata called TTU (https://downloads.teradata.com/download/tools/teradata-tools-and-utilities-linux-installation-package-0)

Proposed operators
* **FastloadOperator**: Wrapper for FastLoad Teradata Utility for loading tables to Teradata
* **FastexportOperator**: Wrapper for Fastexport Teradata Utility for download tables from teradata to CSV
* **BteqOperator**: Wrapper for BTEQ Teradata Utility for BTEQ SQL language

Proposed hooks
* **TtuHook**: Contains all the wrappers with SO processes for executing commands.

**Use case / motivation**

Share and collaborate with people using Teradata and Airflow.

**Questions**
**(1)** Having a dep with Teradata binaries, what considerations should i have to know when contributing with a PR? 
**(2)** I have a first version working in Airflow 1.10; would be better to port as 2.0 hook/op?

Thanks!

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by mik-laj** [comment]
> (1) Having a dep with Teradata binaries, what considerations should i have to know when contributing with a PR?

I recommend that you write a tutorial that describes the necessary steps to use these operators.

> (2) I have a first version working in Airflow 1.10; would be better to port as 2.0 hook/op?

 We do not accept new operators for Airflow 1.10. All new operators are built for Airflow 2.0, but users can use it in Airflow 1.10 thanks to [the backport package](https://github.com/apache/airflow#backport-packages).
